homeassistant/components/vallox/config_flow.py

```diff
@@ -0,0 +1,124 @@
+"""Config flow for the Vallox integration."""
+from __future__ import annotations
+
+from contextlib import suppress
+import ipaddress
+import logging
+import re
+from typing import Any
+
+from vallox_websocket_api import Vallox
+from vallox_websocket_api.exceptions import ValloxApiException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DEFAULT_NAME, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+    }
+)
+
+VALLOX_CONNECTION_EXCEPTIONS = (
+    OSError,
+    ValloxApiException,
+)
+
+
+def host_valid(host: str) -> bool:
+    """Return True if hostname or IP address is valid."""
+    with suppress(ValueError):
+        if ipaddress.ip_address(host).version in [4, 6]:
+            return True
+
+    disallowed = re.compile(r"[^a-zA-Z\d\-]")
+    return all(x and not disallowed.search(x) for x in host.split("."))
+
+
+async def validate_host(hass: HomeAssistant, host: str) -> None:
+    """Validate that the user input allows us to connect."""
+
+    if not host_valid(host):
+        raise InvalidHost(f"Invalid host: {host}")
+
+    client = Vallox(host)
+    await client.get_info()
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for the Vallox integration."""
+
+    VERSION = 1
+
+    async def async_step_import(self, data: dict[str, Any]) -> FlowResult:
+        """Handle import from YAML."""
+        # We need to use the name from the YAML configuration to avoid
+        # breaking existing entity IDs.
+        if CONF_NAME not in data:
+            data[CONF_NAME] = DEFAULT_NAME
+
```

Comment:
```suggestion
```

Reply:
Hm, but the `CONF_NAME` key needs to be present in the `data` dict passed on to `async_create_entry` 🤔

Reply:
You'll still need to merge the data like the other path
https://github.com/home-assistant/core/pull/62780/files#r775610254

Reply:
I had some kind of brainfart here and thought it wouldn't mean the same thing. 🙈 But of course it does 👍 Fixed.